native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java

```diff
@@ -79,6 +69,9 @@ public static Object nextResult(BObject recordIterator, DefaultResultParameterPr
             closeResult(recordIterator);
             return ErrorGenerator.getSQLApplicationError("Error when iterating the SQL result. "
                     + e.getMessage());
+        } catch (BError e) {
+            return ErrorGenerator.getSQLApplicationError("Error when iterating through the " +
+                    "SQL result. " + e.getDetails());
```

Comment:
Rather than handling the error at the top level. Lets catch this at the util class itself and return a FieldMismatchError to the top. It will be much clearer.

Reply:
Fixed in [07ea572](https://github.com/ballerina-platform/module-ballerina-sql/pull/558/commits/07ea57226013d80ea5fa59343a4a6595dd97a6de)